File: app/src/main/java/com/example/shoppingapp/adapters/PreviousOrderAdapter.java
package com.example.shoppingapp.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.shoppingapp.Item;
import com.example.shoppingapp.R;

import java.util.List;


public class PreviousOrderAdapter extends RecyclerView.Adapter<PreviousOrderAdapter.PreviousOrderViewHolder> {

    Context context;
    List<Item> itemList;

    public PreviousOrderAdapter(Context context) {
        this.context = context;
    }

    public PreviousOrderAdapter(Context context, List<Item> itemList) {
        this.context = context;
        this.itemList = itemList;
    }

    @NonNull
    @Override
    public PreviousOrderViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater=LayoutInflater.from(context);
        View v=layoutInflater.inflate(R.layout.previous_order_layout,parent,false);
        return new PreviousOrderAdapter.PreviousOrderViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull PreviousOrderViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 10;
    }

    class PreviousOrderViewHolder extends RecyclerView.ViewHolder{

        ImageView itemImg;
        TextView txtPrice;
        TextView txtName;
        TextView txtQuantity;
        LinearLayout layout;
        public PreviousOrderViewHolder(@NonNull View itemView) {
            super(itemView);

            itemImg = itemView.findViewById(R.id.item_img);
            txtPrice = itemView.findViewById(R.id.txt_price);
            txtName = itemView.findViewById(R.id.txt_name);
            txtQuantity = itemView.findViewById(R.id.txt_quantity);
            layout = itemView.findViewById(R.id.previous_order_layout);
        }
    }
}

File: app/src/customer/java/com/example/shoppingapp_customer/adapters/PreviousOrderAdapter.kt
package com.example.shoppingapp_customer.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.shoppingapp.CartItem
import com.example.shoppingapp.R
import com.example.shoppingapp_customer.adapters.PreviousOrderAdapter.PreviousOrderViewHolder

class PreviousOrderAdapter(var context: Context, var itemList: List<CartItem>) : RecyclerView.Adapter<PreviousOrderViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PreviousOrderViewHolder {
        val v = LayoutInflater.from(context).inflate(R.layout.previous_order_layout, parent, false)
        return PreviousOrderViewHolder(v)
    }

    override fun onBindViewHolder(holder: PreviousOrderViewHolder, position: Int) {
        holder.txtName.text=itemList[position].name
    }
    override fun getItemCount(): Int {
        return itemList.count()
    }

    class PreviousOrderViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val itemImg: ImageView
        val txtPrice: TextView
        val txtName: TextView
        val txtQuantity: TextView


        init {
            itemImg = itemView.findViewById(R.id.img_item_pol)
            txtPrice = itemView.findViewById(R.id.txt_price_pol)
            txtName = itemView.findViewById(R.id.txt_name_dsf)
            txtQuantity = itemView.findViewById(R.id.txt_quantity_dsf)

        }
    }
}